### PR TITLE
Update readme to indicate the road to Cyo

### PR DESCRIPTION
**Pull request by saem**
## Summary
* introduce Cyo into the readme
* hint at the direction of development

### CODE REVIEW

**Review by haxscramper — APPROVED** [review]

File: readme.md [hunk 1]
```diff
@@ -58,17 +63,17 @@ For updates on the progress see the roadmap progress [thread](https://github.com
...
+   (`nkError`) one [Project](https://github.com/nim-works/nimskull/projects/1)
 3. comments - incrementally document compiler source for easier learning
-
-<details><summary><b>Show more</b></summary><br />
+4. slim the core - remove dialects, backwards compatibility, etc [Discussion](https://github.com/nim-works/nimskull/discussions/289)
 
 There are more, the above have been carefully chosen based on the direction of
-the language; moreover, their impact is further beyond than described and the
-intention is to create a virtuous cycle. Examples:
+the language; moreover, their impact is goes beyond what's been described and
```
Comment (zerbina) [inline readme.md:75]:
```suggestion
the language; moreover, their impact goes beyond what's been described and
```

File: readme.md [hunk 2]
```diff
@@ -58,17 +63,17 @@ For updates on the progress see the roadmap progress [thread](https://github.com
 
 The current and key areas of development are as follows:
 
-1. improve tests - clarify the language specification and memorialize it in tests.
+1. improve tests - memorialize the slim core's it in tests.
```
Comment (zerbina) [inline readme.md:66]: This sentence feel off. I'm not sure what the intended meaning is.

Replacing "memorialize" with a more common word or phrase might also be a good idea (in order to make the readme more accessible)

**Review by zerbina — COMMENTED** [review]
Some small things I noticed.

**Comment by saem** [comment]
@zerbina I'll do another pass, thanks for taking the time to review (and Hax, as well).

**Pull request merged**